### PR TITLE
fix: no fallback image and text on picking

### PR DESCRIPTION
**Pull request by tarashaliuk**
closes: [HRZ-3080](https://spryker.atlassian.net/browse/HRZ-3080)

[HRZ-3080]: https://spryker.atlassian.net/browse/HRZ-3080?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 1]
```diff
@@ -29,6 +30,21 @@ describe('PickingComponent', () => {
   let service: MockPickingListService;
   let routerService: MockRouterService;
 
+  const getTabs = () =>
+    element.renderRoot.querySelectorAll('oryx-tab') as NodeListOf<TabComponent>;
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]:
```suggestion
    element.renderRoot.querySelectorAll<TabComponent>('oryx-tab');
```

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 2]
```diff
@@ -29,6 +30,21 @@ describe('PickingComponent', () => {
...
+
+  const onTabs = (
+    tabs: NodeListOf<TabComponent> | TabComponent[],
+    cb: (content: HTMLElement | null) => void
+  ) => {
+    tabs.forEach((tab) => {
+      const tabContent = element.renderRoot.querySelector(
+        `#${tab.getAttribute('for')}`
+      ) as HTMLElement | null;
+      cb(tabContent);
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]:
```suggestion
      const tabContent = element.renderRoot.querySelector<HTMLElement>(
        `#${tab.getAttribute('for')}`
      );
      cb(tabContent);
```

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 3]
```diff
@@ -106,7 +122,27 @@ describe('PickingComponent', () => {
     });
 
     it('should not render product card', () => {
-      expect(element).not.toContainElement('oryx-picking-product-card');
+      onTabs(getTabs(), (tabContent) =>
+        expect(
+          tabContent?.querySelector('oryx-picking-product-card')
+        ).toBeFalsy()
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]:
```suggestion
        expect(
          tabContent!
        ).not.toContainElement('oryx-picking-product-card')
```

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 4]
```diff
@@ -106,7 +122,27 @@ describe('PickingComponent', () => {
...
+          tabContent?.querySelector('.fallback h2')?.textContent
+        ).toContain(i18n('picking.no-items'))
+      );
+    });
+
+    it('should show fallback image', () => {
+      onTabs(getTabs(), (tabContent) =>
+        expect(
+          tabContent?.querySelector('oryx-image[resource="no-orders"]')
+        ).toBeTruthy()
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]:
```suggestion
        expect(
          tabContent!
        ).toContainElement('oryx-image[resource="no-orders"]')
```

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 5]
```diff
@@ -123,15 +159,113 @@ describe('PickingComponent', () => {
...
+          tabContent?.querySelector('.fallback span')?.textContent
+        ).toContain(i18n('picking.all-items-are-processed'));
+      });
+    });
+
+    it('should render success image on "Not Picked" and "NotFound" tabs', () => {
+      onTabs([getTabs()[0], getTabs()[2]], (tabContent) => {
+        expect(
+          tabContent?.querySelector(
+            'oryx-image[resource="picking-items-processed"]'
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]: toContainElement

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 6]
```diff
@@ -123,15 +159,113 @@ describe('PickingComponent', () => {
...
+          )
+        ).toBeTruthy();
+      });
+    });
+
+    it('should render list of picked products on "Picked" tab', () => {
+      onTabs([getTabs()[1]], (tabContent) => {
+        expect(
+          tabContent?.querySelector('oryx-picking-product-card')
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]: toContainElement

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 7]
```diff
@@ -123,15 +159,113 @@ describe('PickingComponent', () => {
...
+          tabContent?.querySelector('.fallback span')?.textContent
+        ).toContain(i18n('picking.all-items-are-processed'));
+      });
+    });
+
+    it('should render success image on "Not Picked" tab', () => {
+      onTabs([getTabs()[0]], (tabContent) => {
+        expect(
+          tabContent?.querySelector(
+            'oryx-image[resource="picking-items-processed"]'
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]: toContainElement

File: libs/domain/picking/src/components/picking/picking.component.spec.ts [hunk 8]
```diff
@@ -123,15 +159,113 @@ describe('PickingComponent', () => {
...
+          )
+        ).toBeTruthy();
+      });
+    });
+
+    it('should render list of picked products on "Picked" and "NotFound" tabs', () => {
+      onTabs([getTabs()[1], getTabs()[2]], (tabContent) => {
+        expect(
+          tabContent?.querySelector('oryx-picking-product-card')
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.spec.ts]: toContainElement

File: libs/domain/picking/src/components/picking/picking.component.ts [hunk 9]
```diff
@@ -236,29 +236,39 @@ export class PickingComponent extends PickingListMixin(LitElement) {
...
-      return;
-    }
-
-    return html`
-      <div class="picking-complete">
-        <div class="img-wrap">
+  protected renderNoItemsFallback(): TemplateResult {
+    if (
+      !this.items?.length ||
+      this.items?.some((item) => item.status !== ItemsFilters.Picked)
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.ts]:
```suggestion
      this.items.some((item) => item.status !== ItemsFilters.Picked)
```

File: libs/domain/picking/src/components/picking/picking.component.ts [hunk 10]
```diff
@@ -236,29 +236,39 @@ export class PickingComponent extends PickingListMixin(LitElement) {
...
+        <div class="fallback">
+          <oryx-heading>
+            <h2>${i18n(`picking.no-items`)}!</h2>
+          </oryx-heading>
+          <oryx-image resource="no-orders"></oryx-image>
+        </div>
+      `;
+    } else {
+      return html`
+        <div class="fallback">
```
Comment (Sinucid) [inline libs/domain/picking/src/components/picking/picking.component.ts]: still a fallback?

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/picking/src/components/picking/picking.component.spec.ts (hunk 3 above)
Comment (tarashaliuk) [reply libs/domain/picking/src/components/picking/picking.component.spec.ts]: `tabContent` is not LitElement so `toContainElement` is not working properly with it

Comment (tarashaliuk) [reply libs/domain/picking/src/components/picking/picking.component.spec.ts]: But I guess we can extend the `toContainElement` method and use it also with HTML elements

Comment (Sinucid) [reply libs/domain/picking/src/components/picking/picking.component.spec.ts]: Oh, I thought that content also a web component, then you can skip this comment

**Review by Sinucid — APPROVED** [review]

**Pull request merged**